[SofaGeneralAnimationLoop] Improvement on MMMapper

olivier-goury opened the pull request:
Quick update of the MechanicalMatrixMapper so that when the user interacts on the simulated object with the mouse, the StiffSpringForceField that is dynamically added to the scene graph gets included in the forcefield link list of the MMMApper.



______________________________________________________
<!--- Please leave this at the end of your message -->
This PR: 
- [x] builds with SUCCESS for all platforms on the CI.
- [x] does not generate new warnings.
- [x] does not generate new unit test failures.
- [x] does not generate new scene test failures.
- [x] does not break API compatibility.
- [x] is more than 1 week old (or has fast-merge label).

**Reviewers will merge only if all these checks are true.**

olivier-goury committed:
Start to correct MMMapper, to cater for the user interaction with the mouse

olivier-goury committed:
Merge branch 'master' into UpdateMMMapper

olivier-goury committed:
Make changes to cater for dynamic update of the list of forcefields, like this happens with mouse interaction

olivier-goury committed:
Remove useless comments

epernod commented:
M&M'sMapper? 🍪 
👍

epernod commented on modules/SofaGeneralAnimationLoop/MechanicalMatrixMapper.h:
```diff
@@ -170,6 +170,8 @@ class SOFA_GENERAL_ANIMATION_LOOP_API MechanicalMatrixMapper : public MixedInter
 
     Data<helper::vector<std::string>> d_forceFieldList;
 
+    unsigned int m_nbInteractionForceFields;
```
Could you use `size_t` instead of unsigned int when you know it will store `stl container size` information.
This avoid warnings of conversion on windows64.

olivier-goury replied:
Thank you. This was corrected.

epernod commented on modules/SofaGeneralAnimationLoop/MechanicalMatrixMapper.inl:
```diff
@@ -318,6 +331,24 @@ void MechanicalMatrixMapper<DataTypes1, DataTypes2>::addKToMatrix(const Mechanic
     time= (double)timer->getTime();
 
+    sofa::simulation::Node *node = l_nodeToParse.get();
+    unsigned int currentNbInteractionFFs = node->interactionForceField.size();
```
same here

olivier-goury replied:
Thank you. This was corrected.

damienmarchal committed:
[SofaGeneralAnimationLoop] Replace loop with foreach

damienmarchal commented:
Olivier's point me that in his code the clearing of the links is really "perfectible".

olivier-goury committed:
Replace unsigned int declarations with size_t and improve clearing of the forceField link list.

olivier-goury commented:
> Olivier's point me that in his code the clearing of the links is really "perfectible".

It was improved with the last commit, thank you!

olivier-goury commented:
Hello!
Why did you change the PR to wip? I think that PR is ready now.

olivier-goury commented:
> Hi @olivier-goury
> The info is in the sofa-dev meeting report but has not been copied here.
> --> wip, use clear() instead of removeAt(0) on l_forceField link
> 
> @guparan will work on it, no worries

Ok thanks for the info!

olivier-goury committed:
Correct bug if the scene has no mass

guparan commented:
Actually there is no clear() method for Links.
